### DISCUSSION TITLE
Using clangd vscode extension instead of c_cpp

### DISCUSSION THREAD

**Discussion by Keyrim** [MaJerle/stm32-cube-cmake-vscode#25] [Ideas]
I just finished your great tutorial and i was just wondering why don't you use [clangd](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd) ?

It uses a [file ](https://clang.llvm.org/docs/JSONCompilationDatabase.html) that must be created during the compilation process which specifies the compilation arguments of each *.c file of the project allowing intellisense to know exactly how each file is compiled resulting therefore in precise warnings & errors.

You no longer need to configure the c_cpp configuration file for each project and cmake [can build this file without any external tool](https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html)

**Comment by MaJerle** [comment]
The c_cpp data relies to a CMake as a data configuration provider, which then generates the necessary intellisense. The C_CPP file is the identical for any project you may have. In fact it is fully supported configuration from within vscodee.

For clangd there is a need to download another server as I see.

At the end, I don't know, I don't find anything problematic with c_cpp file. Works neat.
